common/kafka/kafka_config.cpp

```diff
@@ -0,0 +1,64 @@
+/// Copyright 2019 Pinterest Inc.
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+/// http://www.apache.org/licenses/LICENSE-2.0
+
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+
+#include <fstream>
+
+#include "glog/logging.h"
+#include "common/kafka/kafka_config.h"
+
+namespace kafka {
+
+/**
+ * Read Java kafka client configuration file
+ */
+bool KafkaConfig::read_conf_file(
+  const std::string &conf_file,
+  const std::shared_ptr<ConfigMap> configMap,
+  bool compatibility_flag) {
+  std::ifstream inf(conf_file.c_str());
+
+  if (!inf) {
+    LOG(ERROR) << ": " << conf_file << ": could not open file" << std::endl;
+    return false;
+  }
+
+  LOG(INFO) << ": " << conf_file << ": read config file" << std::endl;
+
+  std::string line;
+  int linenr = 0;
+
+  while (std::getline(inf, line)) {
+    linenr++;
+
+    // Ignore comments and empty lines
+    if (line[0] == '#' || line.length() == 0)
```

Comment:
Should the line.length check be first? line[0] will fail if line is empty.

Reply:
👍